src/components/data/data-table-versions.test.js

```diff
@@ -0,0 +1,145 @@
+      onDismiss: _.noop,
+      onSubmit: _.noop
+    }))
+
+    const setTableCheckboxes = getAllByRole('checkbox')
+    expect(setTableCheckboxes.length).toBe(2)
+
+    expect(setTableCheckboxes[0].parentElement).toHaveTextContent('sample_set')
+    expect(setTableCheckboxes[0].getAttribute('aria-checked')).toBe('false')
+
+    expect(setTableCheckboxes[1].parentElement).toHaveTextContent('sample_set_set')
+    expect(setTableCheckboxes[1].getAttribute('aria-checked')).toBe('false')
+  })
+
+  it('calls onSubmit with entered description and selected set tables', () => {
+    const onSubmit = jest.fn()
+
+    const { getAllByRole, getByLabelText, getByText } = render(h(DataTableSaveVersionModal, {
+      entityType: 'sample',
+      allEntityTypes: ['sample', 'sample_set', 'sample_set_set', 'participant'],
+      onDismiss: _.noop,
+      onSubmit
+    }))
+
+    const descriptionInput = getByLabelText('Description')
+    fireEvent.change(descriptionInput, { target: { value: 'this is a version' } })
+
+    const setTableCheckboxes = getAllByRole('checkbox')
+    fireEvent.click(setTableCheckboxes[0])
+
+    const saveButton = getByText('Save')
+    fireEvent.click(saveButton)
+
+    expect(onSubmit).toHaveBeenCalledWith({
+      description: 'this is a version',
+      includedSetEntityTypes: ['sample_set']
+    })
+  })
+})
+
+describe('DataTableVersion', () => {
+  const testVersion = {
+    url: `gs://workspace-bucket/.data-table-versions/sample/sample.1664568527960.zip`,
+    createdBy: 'user@example.com',
+    entityType: 'sample',
+    includedSetEntityTypes: ['sample_set'],
+    timestamp: 1664568527960,
+    description: 'A version of samples'
+  }
+
+  describe('renders version information', () => {
+    let renderResult
+
+    beforeEach(() => {
+      renderResult = render(h(DataTableVersion, { version: testVersion, onDelete: jest.fn(), onRestore: jest.fn() }))
+    })
+
+    it('renders entity type and timestamp', () => {
+      const { getByRole } = renderResult
+      const heading = getByRole('heading')
+      expect(heading).toHaveTextContent('sample (Sep 30, 2022, 8:08 PM)')
+    })
+
+    it('renders included set tables', () => {
+      const { getByLabelText } = renderResult
+      const setTableList = getByLabelText('Included set tables:')
+      expect(setTableList).toHaveTextContent('sample_set')
+    })
+
+    it('renders creator', () => {
+      const { getByText } = renderResult
+      expect(getByText('Created by: user@example.com')).toBeTruthy()
+    })
+
+    it('renders description', () => {
+      const { getByText } = renderResult
+      expect(getByText('A version of samples')).toBeTruthy()
+    })
+  })
+
+  it('renders restore button and confirms restore', () => {
+    const onRestore = jest.fn()
+    const { getByTestId, getByText } = render(h(DataTableVersion, { version: testVersion, onDelete: jest.fn(), onRestore }))
+
+    const restoreButton = getByText('Import')
```

Comment:
Nit: it might be nice to change "restore" to "import" in the code as well so it isn't left with this legacy of having started as "restore".

Reply:
Done in https://github.com/DataBiosphere/terra-ui/pull/3434/commits/5bd56538295ad92423852dd871b56458ac030230.